src/coreclr/jit/importercalls.cpp

```diff
@@ -5164,12 +5164,16 @@ GenTree* Compiler::impPrimitiveNamedIntrinsic(NamedIntrinsic        intrinsic,
     assert(sig->sigInst.classInstCount == 0);
 
     var_types retType = JITtype2varType(sig->retType);
-    assert(varTypeIsArithmetic(retType));
+
+    if (!varTypeIsArithmetic(retType))
+    {
+        assert((intrinsic == NI_PRIMITIVE_ConvertToInteger) || (intrinsic == NI_PRIMITIVE_ConvertToIntegerNative));
```

Comment:
I am not too familiar with "named intrinsics", but aren't there more intrinsics like `ConvertToInt64` that returns VectorT, although not sure why its entry is not in `namedintrinsiclist.h`. Can you please explain?

Reply:
They are, they are just included indirectly via the tables. That's what this region is for: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/namedintrinsiclist.h#L140-L164

We basically just have 6 categories within the list:
* `NI_SYSTEM_MATH`
* `NI_HW_INTRINSIC`
* `NI_SIMD_AS_HWINTRINSIC`
* `NI_SRCS_UNSAFE`
* `NI_PRIMITIVE`
* general intrinsics not in another list

Each of the first 5 categories are just groups of intrinsics that generally get handled together via some central function for that group.

So the `ConvertToInt64` for `Vector<T>` is handled by `NI_SIMD_AS_HWINTRINSIC` and won't hit this path.